io-engine/src/bin/io-engine.rs

```diff
@@ -92,10 +92,29 @@ fn start_tokio_runtime(args: &MayastorCliArgs) {
 }
 
 fn hugepage_get_nr(hugepage_path: &Path) -> (u32, u32) {
-    let nr_pages: u32 = sysfs::parse_value(hugepage_path, "nr_hugepages")
-        .expect("failed to read the number of pages");
-    let free_pages: u32 = sysfs::parse_value(hugepage_path, "free_hugepages")
-        .expect("failed to read the number of free pages");
+    let nr_pages: u32 = match sysfs::parse_value(hugepage_path, "nr_hugepages") {
+        Ok(c) => c,
+        Err(e) => {
+            warn!(
+                "failed to read the number of pages at {}: {}", hugepage_path.display(), e
+            );
+            // NOTE: We check for 1g pages but the directory for 1g pages won't exist if they are
+            // not enabled. Effectively this means that zero 1g hugepages are available, so it
+            // seems sensible to fall back to 0 if reads fail. See discussion here:
+            // https://github.com/openebs/mayastor/issues/1273.
+            return 0;
```

Comment:
```suggestion
            return (0, 0);
```

Reply:
Good catch, though I suppose this could just be `0` instead of a return? Might as well attempt the free pages lookup in case the error is not a completely missing directory.

Reply:
Yes could be, though chances are it will be no existent too. I've got no strong preference here so I'll leave it up to you :)